File: 2018/Day01/solution.py
freqs = [int(x) for x in open('./input.txt')]

print(f'Part 1: {sum(freqs)}')

seen = set()

current_freq = 0
i = 0

while current_freq not in seen:
    seen.add(current_freq)
    current_freq += freqs[i % len(freqs)]
    i += 1

print(f'Part 2: {current_freq}')

File: 2018/Day01/solution.js
const fs = require('fs');
const data = fs
    .readFileSync('./input.txt', 'utf8')
    .split('\r\n')
    .map(Number);

// Part 1
const resultPart1 = data.reduce((a, c) => a + c);
console.log(resultPart1);

// Part 2
const list = new Map();
let freq = 0;
let i = 0;

while (!list.has(freq)) {
    list.set(freq);
    freq += data[i % data.length];
    i++;
}

console.log(freq);
